call-home/src/bin/stats/main.rs

```diff
@@ -0,0 +1,185 @@
+use clap::Parser;
+
+use crate::{
+    cache::events_cache::{Cache, EventSet},
+    exporter::{events_collector::StatsCollector, exporter_config::ExporterConfig},
+    store::events_store::initialize,
+};
+use actix_web::{http::header, middleware, web, HttpResponse, HttpServer, Responder};
+use k8s_openapi::api::core::v1::ConfigMap;
+use mbus_api::{
+    mbus_nats::{message_bus_init, BusSubscription},
+    message::EventMessage,
+    Bus,
+};
+use obs::common::errors;
+use prometheus::{Encoder, Registry};
+use snafu::ResultExt;
+use std::net::SocketAddr;
+use tracing::{error, info, trace};
+use utils::{
+    raw_version_str,
+    tracing_telemetry::{default_tracing_tags, flush_traces, init_tracing},
+};
+mod cache;
+mod exporter;
+mod store;
+
+#[derive(Parser, Debug)]
+#[clap(name = utils::package_description!(), version = utils::version_info_str!())]
+struct Cli {
+    /// The url for mbus.
+    #[clap(long, default_value = "nats://mayastor-nats:4222")]
+    mbus_url: String,
+
+    /// The namespace we are supposed to operate in.
+    #[arg(short, long, default_value = "mayastor")]
+    namespace: String,
+
+    /// TCP address where events stats are exposed.
+    #[clap(long, short, default_value = "0.0.0.0:9090")]
+    metrics_endpoint: SocketAddr,
+
+    /// Interval to update the config map.
+    #[clap(short, long, default_value = "300s")]
+    update_period: humantime::Duration,
+
+    /// Sends opentelemetry spans to the Jaeger endpoint agent.
+    #[clap(long, short)]
+    jaeger: Option<String>,
+}
+
+impl Cli {
+    fn args() -> Self {
+        Cli::parse()
+    }
+}
+
+/// Intilize mbus.
+pub(crate) async fn mbus_init(mbus_url: &str) -> BusSubscription<EventMessage> {
+    let mut bus = message_bus_init(mbus_url).await;
+    let sub = bus
+        .subscribe::<EventMessage>()
+        .await
+        .map_err(|error| trace!("Error subscribing to jetstream: {error:?}"))
+        .unwrap();
```

Comment:
```suggestion
?;
```

Reply:
This error need to be mapped to the errors::Result ,  but  the returned error is under `common` in mbus_api and this is  a private crate ,, so mapping the mubs::common::errors::Error to common::Error fails .
Making common public in mayastor-dependencies can be done in another PR, if agreed on.